tests/components/lock/test_verisure.py

```diff
@@ -0,0 +1,60 @@
+"""Tests for the Verisure platform."""
+
+from unittest.mock import Mock, patch
+from homeassistant.const import STATE_LOCKED, STATE_UNLOCKED
+
+
+NO_DEFAULT_LOCK_CODE_CONFIG = {
+    'locks': True,
+}
+
+DEFAULT_LOCK_CODE_CONFIG = {
+    'locks': True,
+    'default_lock_code': '9999',
+}
+
+
+async def test_verisure_no_default_code(hass):
+    """Test configs without a default lock code."""
+    with patch('homeassistant.components.lock.verisure.hub') as hub:
+        hub.config = NO_DEFAULT_LOCK_CODE_CONFIG
+
+        from homeassistant.components.lock.verisure import VerisureDoorlock
+        lock = VerisureDoorlock("lock")
+        tests = [
+            {'method': lock.lock, 'state': STATE_LOCKED},
+            {'method': lock.unlock, 'state': STATE_UNLOCKED},
+        ]
+
+        for test in tests:
+            lock.set_lock_state = Mock(name='set_lock_state',
+                                       return_value=None)
+
+            test['method']()
+            lock.set_lock_state.assert_not_called()
```

Comment:
I've been trying to understand what you meant about preventing typos ... did you mean something like the following?
```diff
         for test in tests:
-            lock.set_lock_state = Mock(name='set_lock_state',
-                                       return_value=None)
+            mock = Mock(name='set_lock_state', return_value=None)
+            lock.set_lock_state = mock
 
             test['method']()
-            lock.set_lock_state.assert_not_called()
+            assert mock.call_count == 0
 
             test['method'](code='12345')
-            lock.set_lock_state.assert_called_with('12345', test['state'])
+            assert mock.call_args == call('12345', test['state'])
```

Otherwise I'm not sure I get the point - typos are typos, and it's certainly hard to prevent them on something as magical as python's mocking. I can and will make the change however, just want to make sure I understand the rationale.

Reply:
Yes, exactly like that. Now if we would misspell `mock.call_count` to eg `mock.call_cuont` the test would fail.

Reply:
Makes sense, and https://github.com/home-assistant/home-assistant/pull/18873/commits/c6c927a0857c3e0676ba19a4003aa045f32dbb39 has those changes.